reconcile/utils/ocm/status_board.py

```diff
@@ -0,0 +1,64 @@
+from typing import Any
+
+from reconcile.utils.ocm_base_client import OCMBaseClient
+
+APPLICATION_DESIRED_KEYS = {"id", "name", "fullname", "metadata"}
+PRODUCTS_DESIRED_KEYS = {"id", "name", "fullname", "metadata"}
+
+
+METADATA_MANAGED_BY_KEY = "managedBy"
+METADATA_MANAGED_BY_VALUE = "qontract-reconcile"
+
+
+def get_product_applications(
+    ocm_api: OCMBaseClient, product_id: str
+) -> list[dict[str, Any]]:
+    results: list[dict[str, Any]] = []
+
+    for application in ocm_api.get_paginated(
+        f"/api/status-board/v1/products/{product_id}/applications"
+    ):
+        if (
+            application.get("metadata", {}).get(METADATA_MANAGED_BY_KEY, "")
+            == METADATA_MANAGED_BY_VALUE
+        ):
+            results.append(
+                {k: v for k, v in application.items() if k in APPLICATION_DESIRED_KEYS}
+            )
```

Comment:
Since we are doing fields cherry pick, would it be easier if we define the data model like:

```python
class Application(BaseModel):
    id: Optional[str]
    name: Optional[str]
    fullname: Optional[str]
    metadata: Optional[dict[str, Any]]
```

then it can just be used like

```python
 Application.parse_obj(application)
```

Reply:
@janboll lol, there is a long way to fully typed instead of `dict[str, Any]`